integration_tests/src/main/python/udf_test.py

```diff
@@ -215,17 +215,38 @@ def pandas_filter(iterator):
 
 @pytest.mark.parametrize('data_gen', data_gens_nested_for_udf, ids=idfn)
 def test_pandas_map_udf_nested_type(data_gen):
-    # Spark supports limited types as the return type of pandas UDF.
-    # For details please go to
-    #  https://github.com/apache/spark/blob/master/python/pyspark/sql/pandas/types.py#L28
-    # So we return the data frames with one column of integral type for all types of the input.
-    def size_udf(pdf_itr):
+    # Supported UDF output types by plugin: (commonCudfTypes + ARRAY).nested() + STRUCT
+    # STRUCT represents the whole dataframe in Map Pandas UDF, so no struct column in UDF output.
+    # More details is here
+    #   https://github.com/apache/spark/blob/master/python/pyspark/sql/udf.py#L119
```

Comment:
That code seems to only indicate that the top-level type is expected to be a struct, not that it doesn't support struct-in-struct?  Poking around at Spark Pandas UDF examples, it looks like struct columns are supported as a result column.

Reply:
I verified it locally, and it failed when returning a struct column (even as a top-level column) from Map Pandas UDF.
I uploaded the test code to my repo, you can find it here: https://github.com/firestarman/spark-rapids/commit/02b5b4d4e4a2547fcc35cece04f79d7f34cc70b2 
```bash
        elif self.evalType == PythonEvalType.SQL_MAP_PANDAS_ITER_UDF:
            if isinstance(self._returnType_placeholder, StructType):
                try:
                    to_arrow_type(self._returnType_placeholder)
                except TypeError:
                    raise NotImplementedError(
                        "Invalid return type in mapInPandas: "
>                       "%s is not supported" % str(self._returnType_placeholder))
E                   NotImplementedError: Invalid return type in mapInPandas: StructType(List(StructField(a,StructType(List(StructField(child0,ArrayType(ShortType,true),true),StructField(child1,StructType(List(StructField(child0,ByteType,true),StructField(child1,ShortType,true),StructField(child2,IntegerType,true),StructField(child3,LongType,true),StructField(child4,FloatType,true),StructField(child5,DoubleType,true),StructField(child6,StringType,true),StructField(child7,BooleanType,true),StructField(child8,DateType,true),StructField(child9,TimestampType,true))),true))),true))) is not supported
```

Whether it supports non-nested struct column depends on the type of Pandas UDF, I think. 
e.g. the Scalar Pandas UDF should support top-level struct column, but Map Pandas UDF does not. Since the former one does not require the top-level type being a struct, while the later one does. 
Actually Map Pandas UDF requires returning a DataFrame from UDF, not a column. And the type of a DataFrame is `struct`.